File: app/src/main/java/codes/umair/rgbwallpaper/Fragments/RGBWallaperFrag.java
package codes.umair.rgbwallpaper.Fragments;


import android.Manifest;
import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.SeekBar;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;
import com.nabinbhandari.android.permissions.PermissionHandler;
import com.nabinbhandari.android.permissions.Permissions;

import java.io.IOException;
import java.util.ArrayList;

import codes.umair.rgbwallpaper.R;
import codes.umair.rgbwallpaper.Util;

/**
 * Created by Umair Ayub on 8/08/2019.
 */

public class RGBWallaperFrag extends Fragment {
    
    private SeekBar seekR;
    private SeekBar seekG;
    private SeekBar seekB;
	private TextView textCode;
	private FloatingActionButton fabRan,fabChng,fabSave;
	private ImageView previewV;
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.rgb_frag,container,false);
       // buttonRandom = (Button) view.findViewById(R.id.mainButtonRandom);
		fabRan = view.findViewById(R.id.mainButtonRandom);
		fabChng = view.findViewById(R.id.fab);
		fabSave = view.findViewById(R.id.fabsave);
		seekR = view.findViewById(R.id.mainSeekBarR);
		seekG = view.findViewById(R.id.mainSeekBarG);
		seekB = view.findViewById(R.id.mainSeekBarB);
		textCode = view.findViewById(R.id.mainTextViewCode);
		previewV = view.findViewById(R.id.previewV);
		GenerateRandom();

		seekR.setOnSeekBarChangeListener(new displayRGB());
        seekG.setOnSeekBarChangeListener(new displayRGB());
        seekB.setOnSeekBarChangeListener(new displayRGB());


		fabSave.setOnClickListener(new OnClickListener(){

				@Override
				public void onClick(final View p1)
				{
					// TODO: Implement this method
					
					final Bitmap bitmap = Util.createColorBitmap(getActivity(),Color.rgb(seekR.getProgress(),seekG.getProgress(),seekB.getProgress()));
					String rationale = "Please provide Storage permission to save Wallpapers.";
					String[] permissions = {Manifest.permission.WRITE_EXTERNAL_STORAGE};
					
					Permissions.Options options = new Permissions.Options()
						.setRationaleDialogTitle("Info")
						.setSettingsDialogTitle("Warning");

					Permissions.check(getActivity(), permissions, rationale, options, new PermissionHandler() {
							@Override
							public void onGranted() {
								// do your task.
								Util.SaveBitmap(p1,bitmap);
							}

							@Override
							public void onDenied(Context context, ArrayList<String> deniedPermissions) {
								// permission denied, block the feature.
							}
						});
					
				}


			});
		fabRan.setOnClickListener(new OnClickListener(){

				@Override
				public void onClick(View p1)
				{
					// TODO: Implement this method
					GenerateRandom();
				}
			});
		fabChng.setOnClickListener(new OnClickListener(){

				@Override
				public void onClick(View p1)
				{
					// TODO: Implement this method
					try{
						Util.setWallpaper(getActivity(),Util.createColorBitmap(getActivity(),Color.rgb(seekR.getProgress(),seekG.getProgress(),seekB.getProgress())));
						Snackbar.make(p1,"Wallpaper Changed Successfully!",Snackbar.LENGTH_LONG).show();
					}catch (IOException e){
						e.printStackTrace();
						Snackbar.make(p1, "Unknown Error Occurred while changing wallpaper!", Snackbar.LENGTH_LONG).show();
					}
					
				}


			});
        return view;
    }
	
	
	public void GenerateRandom(){
		int randomR = Util.Random();
		int randomG = Util.Random();
		int randomB = Util.Random();

		seekR.setProgress(randomR);
		seekG.setProgress(randomG);
		seekB.setProgress(randomB);

		previewV.setBackgroundColor(Color.rgb(randomR, randomG, randomB));
	}


	@Override
    public void onResume(){
        super.onResume();

        new displayRGB().onProgressChanged(seekR, 1, true);

    }

	

	public class displayRGB implements SeekBar.OnSeekBarChangeListener {

        @Override
        public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {

            previewV.setBackgroundColor(Color.rgb(seekR.getProgress(), seekG.getProgress(), seekB.getProgress()));

            String red = Integer.toHexString(seekR.getProgress());
            red = Util.zero(red);

            String green = Integer.toHexString(seekG.getProgress());
            green = Util.zero(green);

            String blue = Integer.toHexString(seekB.getProgress());
            blue = Util.zero(blue);

            String color = "#" + red + green + blue;

            textCode.setText(color);
			textCode.setTextColor(Color.rgb(
									  seekR.getMax() - seekR.getProgress(),
									  seekG.getMax() - seekG.getProgress(),
									  seekB.getMax() - seekB.getProgress() ));


        }

        @Override
        public void onStartTrackingTouch(SeekBar seekBar) {

        }

        @Override
        public void onStopTrackingTouch(SeekBar seekBar) {

        }
    }
	
}

File: app/src/main/java/codes/umair/rgbwallpaper/fragments/RGBWallaperFrag.kt
package codes.umair.rgbwallpaper.fragments

import android.Manifest
import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Context.CLIPBOARD_SERVICE
import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import android.widget.TextView
import androidx.fragment.app.Fragment
import codes.umair.rgbwallpaper.R
import codes.umair.rgbwallpaper.Util
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.material.snackbar.Snackbar
import com.nabinbhandari.android.permissions.PermissionHandler
import com.nabinbhandari.android.permissions.Permissions
import java.io.IOException
import java.util.*

/**
 * Created by Umair Ayub on 8/08/2019.
 */
class RGBWallaperFrag : Fragment() {

    private var seekR: SeekBar? = null
    private var seekG: SeekBar? = null
    private var seekB: SeekBar? = null
    private var textCode: TextView? = null
    private var fabRan: FloatingActionButton? = null
    private var fabChng: FloatingActionButton? = null
    private var fabSave: FloatingActionButton? = null
    private var root: RelativeLayout? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.rgb_frag, container, false)

        fabRan = view.findViewById(R.id.mainButtonRandom)
        fabChng = view.findViewById(R.id.fab)
        fabSave = view.findViewById(R.id.fabsave)
        seekR = view.findViewById(R.id.mainSeekBarR)
        seekG = view.findViewById(R.id.mainSeekBarG)
        seekB = view.findViewById(R.id.mainSeekBarB)
        textCode = view.findViewById(R.id.mainTextViewCode)
        root = view.findViewById(R.id.root)

        generateRandom()

        seekR?.setOnSeekBarChangeListener(DisplayRGB())
        seekG?.setOnSeekBarChangeListener(DisplayRGB())
        seekB?.setOnSeekBarChangeListener(DisplayRGB())

        textCode?.setOnClickListener {
            //Copy Color Code
            val myClipboard: ClipboardManager? = context?.getSystemService(CLIPBOARD_SERVICE) as ClipboardManager?
            val myClip = ClipData.newPlainText("text", textCode?.text)
            myClipboard?.setPrimaryClip(myClip)
            Snackbar.make(it, "Copied to Clipboard!", Snackbar.LENGTH_LONG).show()

        }

        fabSave?.setOnClickListener { it ->
            val bitmap = Util().createColorBitmap(activity!!, Color.rgb(seekR!!.progress, seekG!!.progress, seekB!!.progress))
            val rationale = "Please provide Storage permission to save Wallpapers."
            val permissions = arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE)
            val options = Permissions.Options()
                    .setRationaleDialogTitle("Info")
                    .setSettingsDialogTitle("Warning")
            Permissions.check(activity, permissions, rationale, options, object : PermissionHandler() {
                override fun onGranted() {
                    // do your task.
                    Util().saveBitmap(it, bitmap)
                }

                override fun onDenied(context: Context, deniedPermissions: ArrayList<String>) {
                    // permission denied, block the feature.
                }
            })
        }
        fabRan?.setOnClickListener { generateRandom() }
        fabChng?.setOnClickListener { it ->
            try {
                Util().setWallpaper(activity, Util().createColorBitmap(activity!!, Color.rgb(seekR!!.progress, seekG!!.progress, seekB!!.progress)))
                Snackbar.make(view, "Wallpaper Changed Successfully!", Snackbar.LENGTH_LONG).show()
            } catch (e: IOException) {
                e.printStackTrace()
                Snackbar.make(it, "Unknown Error Occurred while changing wallpaper!", Snackbar.LENGTH_LONG).show()
            }
        }
        return view
    }

    private fun generateRandom() {
        val randomR = Util().random()
        val randomG = Util().random()
        val randomB = Util().random()
        seekR!!.progress = randomR
        seekG!!.progress = randomG
        seekB!!.progress = randomB
        root!!.setBackgroundColor(Color.rgb(randomR, randomG, randomB))

    }

    override fun onResume() {
        super.onResume()
        DisplayRGB().onProgressChanged(seekR!!, 1, true)
    }

    inner class DisplayRGB : OnSeekBarChangeListener {
        override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
            root!!.setBackgroundColor(Color.rgb(seekR!!.progress, seekG!!.progress, seekB!!.progress))
            var red = Integer.toHexString(seekR!!.progress)
            red = Util().zero(red)
            var green = Integer.toHexString(seekG!!.progress)
            green = Util().zero(green)
            var blue = Integer.toHexString(seekB!!.progress)
            blue = Util().zero(blue)
            val color = "#$red$green$blue"
            textCode!!.text = color
        }

        override fun onStartTrackingTouch(seekBar: SeekBar) {}
        override fun onStopTrackingTouch(seekBar: SeekBar) {}
    }
}